### PR TITLE
CodeArena: Style and Gas

### PR DESCRIPTION
**Pull request by MikeHathaway**
# Description of change
## High level
* Various style and gas improvements
  * [Simplify typecasting](https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/pontifex-G.md#gas-3-standardfundingsol_redundant-type-casting-in-the-loop)
  * [remove redundant variable from _validateCalldatas](https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/pontifex-G.md#gas-4-fundingsoll130_redundant-caching-variable)
  * [Remove redundant overflow check from _fundingVote](https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/rbserver-Q.md#11-redundant-execution-of-if-sumofthesquareofvotescast--typeuint128max-revert-insufficientvotingpower-in-standardfunding_fundingvote-function)
  * [Use named variables for mappings](https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/nadin-Q.md)
  * Fix typos and spelling errors
  * [Expand natspec](https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/BRONZEDISC-Q.md#natspec-missing-3)

# Description of bug or vulnerability and solution
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/lukris02-Q.md
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/nadin-Q.md
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/nzm_-G.md
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/rbserver-Q.md
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/pontifex-G.md
* https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/BRONZEDISC-Q.md

# Gas usage
Slight decrease in gas usage, especially on `fundingVote`, `screeningVote`, and `propose`.
## Pre Change
```
| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |                                                                                                             
|---------------------------------------------|-----------------|--------|--------|--------|---------|                                                                                                             
| Deployment Cost                             | Deployment Size |        |        |        |         |                                                                                                             
| 3337342                                     | 16676           |        |        |        |         |                                                                                                             
| Function Name                               | min             | avg    | median | max    | # calls |                                                                                                             
| claimDelegateReward                         | 645             | 56571  | 58435  | 68035  | 247     |                                                                                                             
| execute                                     | 8843            | 32221  | 37580  | 47334  | 10      |                                                                                                             
| fundTreasury                                | 8178            | 41840  | 44818  | 65765  | 25      |                                                                                                             
| fundingVote                                 | 1627            | 107240 | 107458 | 416313 | 265     |                                                                                                             
| getDelegateReward                           | 3117            | 3607   | 3117   | 5159   | 5       |                                                                                                             
| getDescriptionHash                          | 894             | 918    | 930    | 930    | 86      |                                                                                                             
| getDistributionId                           | 435             | 491    | 435    | 2435   | 640     |                                                                                                             
| getDistributionPeriodInfo                   | 1088            | 1830   | 1088   | 5088   | 97      |                                                                                                             
| getFundedProposalSlate                      | 1154            | 1242   | 1154   | 1389   | 8       |                                                                                                             
| getFundingVotesCast                         | 1601            | 2341   | 2341   | 3081   | 2       |                                                                                                             
| getProposalInfo                             | 1000            | 1000   | 1000   | 1000   | 121     |                                                                                                             
| getSlateHash                                | 911             | 919    | 923    | 923    | 3       |                                                                                                             
| getStage                                    | 1485            | 2314   | 1591   | 9591   | 21      |                                                                                                             
| getTopTenProposals                          | 1187            | 2351   | 2363   | 3304   | 20      |                                                                                                             
| getVoterInfo                                | 1034            | 1034   | 1034   | 1034   | 5       |                                                                                                             
| getVotesFunding                             | 2439            | 6639   | 2847   | 11985  | 15      |                                                                                                             
| getVotesScreening                           | 5156            | 5164   | 5156   | 6137   | 503     |                                                                                                             
| hashProposal                                | 4029            | 4029   | 4029   | 4029   | 76      |                                                                                                             
| propose                                     | 2936            | 77196  | 83745  | 83745  | 88      |                                                                                                             
| screeningVote                               | 1470            | 44642  | 37348  | 397982 | 285     |                                                                                                             
| screeningVotesCast                          | 675             | 1341   | 675    | 2675   | 3       |                                                                                                             
| startNewDistributionPeriod                  | 571             | 36512  | 36196  | 52240  | 28      |                                                                                                             
| state                                       | 1131            | 2024   | 2482   | 2777   | 10      |                                                                                                             
| treasury                                    | 396             | 396    | 396    | 396    | 8       |                                                                                                             
| updateSlate                                 | 1160            | 57120  | 70111  | 317852 | 20      |   
```
## Post Change
```
| src/grants/GrantFund.sol:GrantFund contract |                 |        |        |        |         |                                                                                                             
|---------------------------------------------|-----------------|--------|--------|--------|---------|                                                                                                             
| Deployment Cost                             | Deployment Size |        |        |        |         |                                                                                                             
| 3284478                                     | 16412           |        |        |        |         |                                                                                                             
| Function Name                               | min             | avg    | median | max    | # calls |                                                                                                             
| claimDelegateReward                         | 645             | 56874  | 58435  | 68035  | 295     |                                                                                                             
| execute                                     | 8843            | 32221  | 37580  | 47334  | 10      |                                                                                                             
| fundTreasury                                | 8178            | 41840  | 44818  | 65765  | 25      |                                                                                                             
| fundingVote                                 | 1209            | 107001 | 107199 | 415608 | 313     |                                                                                                             
| getDelegateReward                           | 3117            | 3607   | 3117   | 5159   | 5       |                                                                                                             
| getDescriptionHash                          | 894             | 916    | 930    | 930    | 93      |                                                                                                             
| getDistributionId                           | 435             | 483    | 435    | 2435   | 743     |                                                                                                             
| getDistributionPeriodInfo                   | 1088            | 1780   | 1088   | 5088   | 104     |                                                                                                             
| getFundedProposalSlate                      | 1154            | 1242   | 1154   | 1389   | 8       |                                                                                                             
| getFundingVotesCast                         | 1601            | 2341   | 2341   | 3081   | 2       |                                                                                                             
| getProposalInfo                             | 1000            | 1000   | 1000   | 1000   | 121     |                                                                                                             
| getSlateHash                                | 911             | 919    | 923    | 923    | 3       |                                                                                                             
| getStage                                    | 1485            | 2314   | 1591   | 9591   | 21      |                                                                                                             
| getTopTenProposals                          | 1187            | 2351   | 2363   | 3304   | 20      |                                                                                                             
| getVoterInfo                                | 1034            | 1034   | 1034   | 1034   | 5       |                                                                                                             
| getVotesFunding                             | 2439            | 6639   | 2847   | 11985  | 15      |                                                                                                             
| getVotesScreening                           | 5156            | 5162   | 5156   | 6137   | 599     |                                                                                                             
| hashProposal                                | 4029            | 4029   | 4029   | 4029   | 83      |                                                                                                             
| propose                                     | 2936            | 77517  | 82972  | 83649  | 95      |                                                                                                             
| screeningVote                               | 1052            | 44567  | 36802  | 395209 | 333     |                                                                                                             
| screeningVotesCast                          | 675             | 1341   | 675    | 2675   | 3       |                                                                                                             
| startNewDistributionPeriod                  | 571             | 36443  | 36122  | 52166  | 28      |                                                                                                             
| state                                       | 1131            | 2024   | 2482   | 2777   | 10      |                                                                                                             
| treasury                                    | 396             | 396    | 396    | 396    | 8       |                                                                                                             
| updateSlate                                 | 1160            | 57120  | 70111  | 317852 | 20      |    
```

### CODE REVIEW

File: src/grants/GrantFund.sol [hunk 1]
```diff
@@ -384,6 +384,7 @@ contract GrantFund is IGrantFund, Storage, ReentrancyGuard {
         uint256 noOfCalldatas = calldatas_.length;
         for (uint256 i = 0; i < noOfCalldatas;) {
             // proposals can only ever target the Ajna token contract, with 0 value
+            // TODO: https://github.com/code-423n4/2023-05-ajna-findings/blob/main/data/Sathish9098-Q.md#l-5-gas-griefingtheft-is-possible-on-unsafe-external-call
```
Comment (grandizzy) [inline src/grants/GrantFund.sol]: will this be addressed?

File: src/grants/base/Storage.sol [hunk 2]
```diff
@@ -66,50 +66,50 @@ abstract contract Storage is IGrantFundState {
      * @notice Mapping of distribution periods from the grant fund.
      * @dev distributionId => DistributionPeriod
      */
-    mapping(uint24 => DistributionPeriod) internal _distributions;
+    mapping(uint24 distributionId => DistributionPeriod) internal _distributions;
```
Comment (grandizzy) [inline src/grants/base/Storage.sol:69]: neat

**Review by grandizzy — APPROVED** [review]
LGTM, only comment re TODO

File: src/grants/GrantFund.sol (hunk 1 above)
Comment (MikeHathaway) [reply src/grants/GrantFund.sol]: I don't think this is a problem for us - but added that TODO to highlight for reviews and second opinions. Can remove it and make a card.

Comment (grandizzy) [reply src/grants/GrantFund.sol]: not sure what's the best way to get confirmation on it, leaving up to you...

Comment (MikeHathaway) [reply src/grants/GrantFund.sol]: I created this card to track: https://www.notion.so/worlds-best-crypto-lending-platform/Investigate-_execute-QA-Issue-in-Grant-Fund-57ec6e77a483437385d132a06174d9f6?pvs=4

**Review by mattcushman — APPROVED** [review]
lgtm

**Pull request merged**